src/modules/todoList.js

```diff
@@ -7,22 +9,44 @@ export default class TodoList {
   }
 
   addTask(description, completed = false) {
-    const newTask = new Task(description, completed, this.newIndex());
+    const newIndex = this.tasks.length + 1;
+    const newTask = new Task(description, completed, newIndex);
     this.tasks.push(newTask);
     this.drawTable();
+    Form.refreshTasksEvents();
+    LocalStorage.saveToLocalStorage(this.tasks);
   }
 
-  sortedTasks() {
-    return this.tasks.sort((a, b) => a.index - b.index);
+  removeTask(indexToRemove) {
+    let found = false;
+
+    this.tasks.forEach((task, index) => {
+      if (task.index === indexToRemove) {
+        this.tasks.splice(index, 1);
+        found = true;
+      }
+
+      if (found && indexToRemove <= this.tasks.length) {
+        this.tasks[index].index -= 1;
+      }
+    });
+    LocalStorage.saveToLocalStorage(this.tasks);
   }
```

Comment:
- [ ] I experienced an unexpected behavior from your application while removing a todo. Kindly follow these steps to see what I experience from your application :thinking:

1. Add four todos :blush:
2. After adding todos refresh your page using the refresh button on your browser or by pressing **F5** on your keyboard :+1:
3. After refreshing your page once kindly remove the second last todo (3rd todo because in this case, it is the second last todo)

You will see that two todos will be removed just by removing the second last (3rd todo) and 2 todos will be remaining out of 4 todos. Kindly fix this issue :+1:

Reply:
Hi Saleem,

Thank you very much for your detailed report. I was adding two times the event listener for the delete button.